src/chainstate/stacks/db/blocks.rs

```diff
@@ -3104,12 +3104,36 @@ impl StacksChainState {
         return Some((end, None));
     }
 
-    /// Validate an anchored block against the burn chain state.
+    /// Determine whether or not a block executed an epoch transition.  That is, did this block
+    /// call `initialize_epoch_2_05()` or similar when it was processed.
+    pub fn block_crosses_epoch_boundary(
+        block_conn: &DBConn,
+        parent_consensus_hash: &ConsensusHash,
+        parent_block_hash: &BlockHeaderHash,
+    ) -> Result<bool, db_error> {
+        let sql = "SELECT 1 FROM epoch_transitions WHERE block_id = ?1";
+        let args: &[&dyn ToSql] = &[&StacksBlockHeader::make_index_block_hash(
+            parent_consensus_hash,
+            parent_block_hash,
+        )];
+        let res = block_conn
+            .query_row(sql, args, |_r| Ok(()))
+            .optional()
+            .map(|x| x.is_some())?;
+
+        Ok(res)
+    }
```

Comment:
Can you add an assertion to one of the `stacks-node::tests::epoch_205` tests that asserts this function returns the expected boolean at the appropriate block?

Reply:
Yup, added